app/src/main/java/com/fcaputo/parkingapp/mvp/view/SettingsView.kt

```diff
@@ -0,0 +1,76 @@
+package com.fcaputo.parkingapp.mvp.view
+
+import android.app.Activity
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import com.fcaputo.parkingapp.R
+import com.fcaputo.parkingapp.databinding.ActivitySettingsBinding
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.mvp.view.base.ActivityView
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class SettingsView(activity: Activity) : ActivityView(activity),SettingsContract.View {
+    private var binding: ActivitySettingsBinding = ActivitySettingsBinding.inflate(activity.layoutInflater)
+
+    init {
+        activity.setContentView(binding.root)
+    }
+
+    override fun onSaveButton(onClick: () -> Unit) {
+        binding.saveButton.setOnClickListener{
+            if(fieldsAreComplete()) {
+                onClick()
+            } else {
+                showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            }
+        }
+    }
+
+    private fun fieldsAreComplete(): Boolean = listOf(binding.tInputParkingSize).all{ it.editText?.text?.isNotEmpty() ?: false }
+
+    override fun clear() { binding.tInputParkingSize.editText?.text?.clear() }
+
+    override fun getParkingSize(): Int = binding.tInputParkingSize.editText?.text.toString().toInt()
```

Comment:
If the `.toString()` returns "null", the `.toInt()` may arise an exception. Maybe you could return:
`binding.tInputParkingSize.editText?.text.toInt() ?: -1`
Being -1 the error case. Or even:
`override fun getParkingSize(): Int? = binding.tInputParkingSize.editText?.text?.toIntOrNull()`
And handle that nullability in the presenter

Reply:
I prefer the second one, btw

Reply:
Agreed, the second option is better

Reply:
I should include the toString() method anyway, the IDE tells me that toIntOrNull() doesn't work directly with Editable (the type of .text property)

<img width="330" alt="image" src="https://github.com/CFedericoDavid/g-reskilling-android-parkingApp/assets/13878290/48fc2caa-338a-4973-b2b1-920c708c8398">

Reply:
How could it return null if the UI is restricted to write valid strings into the inputs?

Reply:
Yeah, I believe you're right. In practice it shouldn't be null, as the child layout of your `TextInputLayout` is a descendant of an `EditText`.